### PR TITLE
Rename guaranteed to min and ceiling to max

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind cleanup
/kind api-change

#### What this PR does / why we need it:

The `guaranteed` and `ceiling` fields where named such for lack of better names.

`guranteed` is highly misleading, because without preemption and scheduler enforcement, we can't necessarily guarantee the quota. `min` seems less problematic. Then `max` makes more sense than `ceiling`.

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: api/v1alpha1/clusterqueue_types.go [hunk 1]
```diff
@@ -189,48 +170,17 @@ type Resource struct {
...
+	//       min: 10
 	//
 	// The flavors are evaluated in order, selecting the first to satisfy a
 	// workload’s requirements. Also the quantities are additive, in the example
 	// above the GPU quota in total is 20 (10 k80 + 10 p100).
 	// A workload is limited to the selected type by converting the labels to a node
-	// selector that gets injected into the workload. ​​This list can’t be empty, at
-	// least one must exist.
-	//
-	// Note that a workload’s node affinity/selector constraints are evaluated
```
Comment (ahg-g) [inline api/v1alpha1/clusterqueue_types.go:208]: It is worth having this in the docs

File: docs/concepts/cluster_queue.md [hunk 2]
```diff
@@ -177,9 +177,10 @@ When borrowing, Kueue satisfies the following semantics:
 
 - When assigning flavors, Kueue goes through the list of flavors in
   `.spec.requestableResources[*].flavors`. For each flavor, Kueue attempts to
-  fit the workload using the guaranteed quota of the ClusterQueue or the unused
-  quota in the cohort. If the workload doesn't fit, Kueue proceeds evaluating
-  the next flavor in the list.
+  fit the workload using the min quota of the ClusterQueue or the unused
+  min quota of other ClusterQueues in the cohort, up to the max quota of the
+  ClusterQueue. If the workload doesn't fit, Kueue proceeds evaluating the next\
```
Comment (ahg-g) [inline docs/concepts/cluster_queue.md:182]:
```suggestion
  ClusterQueue. If the workload doesn't fit, Kueue proceeds evaluating the next
```

File: api/v1alpha1/clusterqueue_types.go (hunk 1 above)
Comment (alculquicondor) [reply api/v1alpha1/clusterqueue_types.go:208]: It is https://github.com/kubernetes-sigs/kueue/blob/main/docs/concepts/cluster_queue.md#resourceflavor-labels

File: docs/concepts/cluster_queue.md (hunk 2 above)
Comment (alculquicondor) [reply docs/concepts/cluster_queue.md]: Oops

**Comment by alculquicondor** [comment]
Oops, forgot to add a timeout to the ExpectWorkloadsToBePending check. It seems that the gomega default is 1s, which might not be enough for the slow prow workers.

**Pull request merged**